aoc2022/src/main/scala/day15.scala

```diff
@@ -0,0 +1,198 @@
+ *   │          .###.   .     . .               │
+ *   │         .#####.   .   .   .              │
+ *   │  .     .#######.   . .     .             │
+ *   │   .   .#########.   X       .            │
+ *   │    . .###########. . .       .           │
+ *   │     D######S######B   .       .          │
+ *   │    . .###########.     .     .#.         │
+ *   │   .   .#########.       .   .###.   .    │
+ *   │  .     .#######.         . .#####. .     │
+ *   │         .#####.           .#######.      │
+ *   │          .###.             .#####. .     │
+ *   │           .#.               .###.   .    │
+ *   │            C                 .#.         │
+ *   │           . .                 .          │
+ *   │          .   .               . .         │
+ *   │         .     .             .   .        │
+ *   │                                          │
+ *   └──────────────────────────────────────────┘
+ * 
+ *            1         2
+ *   0        0         0   x
+ *  0┌────────────────────────┐
+ *   │S########S#############S│
+ *   │#############R####B#####│
+ *   │########################│
+ *   │########################│
+ *   │#####B##################│
+ *   │##################F#####│
+ *   │##############.#########│
+ *   │########################│
+ *   │########################│
+ * 10│########W######Q########│
+ *   │#################B######│
+ *   │##B#####################│
+ *   │########################│
+ *   │########################│
+ *  y│S######################S│
+ *   └────────────────────────┘
+ * 
+ * The solution involves checking for the computed line intersections XY coordinates if it's within the range of a 
+ * sensor. If not, then that must be the distress beacon. 
+ *
+ */
+
+
+object day15 extends App:
+
+  private val day: String =
+    this.getClass.getName.drop(3).init
+
+  private val start1: Long =
+    System.currentTimeMillis
+
+  private val input: Vector[Sensor] =
+    
+    def parseSensor(s: String): Sensor = s match
+      case s"Sensor at x=$x, y=$y: closest beacon is at x=$bx, y=$by" => 
+        val sensor: Point = Point(x.toInt, y.toInt)
+        val beacon: Point = Point(bx.toInt, by.toInt)
+        Sensor(sensor, beacon, sensor.manhattan(beacon))
+      case _ => sys.error(s"Couldn't parse sensor: $s")
+    
+    Source
+      .fromResource(s"day$day.txt")
+      .getLines
+      .map(parseSensor)
+      .toVector
+    
+  case class Sensor(loc: Point, beacon: Point, innerDist: Int)
+  
+  extension (p: Point) 
+    def manhattan(that: Point): Int = math.abs(p.x - that.x) + math.abs(p.y -  that.y)
+  
+  object Sensor: 
+    
+    def distressBeacon(ss: Vector[Sensor], in: Point): Option[Point] =
+      val withinRange: Boolean = ss.exists((s: Sensor) => s.loc.manhattan(in) <= s.innerDist)
+      if withinRange then None else Some(in)
+      
+    def sensorEnds(s: Sensor, spacing: Int = 0): Vector[Point] =
+      val d: Int = s.innerDist + spacing
```

Comment:
What was your idea with the `spacing`? It seems to always be `0` :wink: Then it might be better to just remove it.

Reply:
The spacing determines the extra distance that's added to the sensor edge coordinate. E.g.: 

```


         .                .
        ...              ...
       ..#..            .....
      ..###..          ...#...
     ..#####..        ...###...
      ..###..        ...#####...
       ..#..          ...###...
        ...            ...#...
         .              .....
                         ...
                          .

``` 
Above two sensors with spacing 2 and 3, for example